server/drafty/drafty.go

```diff
@@ -19,6 +19,62 @@ type span struct {
 	data map[string]interface{}
 }
 
+func (s *span) fromMap(in interface{}) error {
+	m, _ := in.(map[string]interface{})
+	if m == nil {
+		return errUnrecognizedContent
+	}
+
+	s.tp, _ = m["tp"].(string)
+	s.at = intFromNumeric(m["at"])
+	s.end = s.at + intFromNumeric(m["len"])
+	if s.end < s.at {
+		return errInvalidContent
+	}
+	if s.tp == "" {
+		s.key = intFromNumeric(m["key"])
+	}
+
+	return nil
+}
+
+func intFromNumeric(num interface{}) int {
+	switch i := num.(type) {
+	case int:
+		return i
+	case int16:
+		return int(i)
+	case int32:
+		return int(i)
+	case int64:
+		return int(i)
+	case float32:
+		return int(i)
+	case float64:
+		return int(i)
+	default:
+		return 0
```

Comment:
Should we fail here? E.g. return an error, or at least log an error so we know there's a problem.

Alternatively, I suggest changing the function signature to something like
`intFromNumeric(num interface{}, default int) int`
and return the `default` value in case `num` is undefined.

Reply:
Yes, you are right. Let's fail here.